### DISCUSSION TITLE
Trying to avoid false alerts of degraded health

### DISCUSSION THREAD

**Discussion by mubarak-j** [argoproj/argo-cd#7936] [Q&A] [answered]
When using HPA and deploying apps, there's a short period of time until HPA collects metrics e.g CPU and the pod is fully healthy and CPU is collected.

Meanwhile, ArgoCD marks the app in degraded health status and notifications about degraded health are sent. Is there a way in argocd or argocd-notifications where can delay the report of health status until pod is ready or even simply applying a fixed wait time?

**Comment by RomanBats** [comment]
I am not an argocd-notifications expert, but I guess it is possible to customize triggers with some timing conditions.
There is an example for something like this on [triggers page](https://argocd-notifications.readthedocs.io/en/stable/triggers/#functions):
```yaml
when: time.Now().Sub(time.Parse(app.status.operationState.startedAt)).Minutes() >= 5
```

**Reply by prein** [reply]
If my understanding is correct, it would work if one was able to do something along the lines of `ApplicationStatus.HealthStatus is "degraded" and ApplicationStatus.startedAt more than N minutes ago`. The problem is ApplicationStatus doesn't have startedAt (or whatever we could name it) - a property holding the time when the status have changed).

**Reply by mubarak-j** [reply, marked as answer]
I end up using [resource customizations](https://argo-cd.readthedocs.io/en/stable/operator-manual/health/#way-1-define-a-custom-health-check-in-argocd-cm-configmap) to define custom health check for HPA resources:

```
    resource.customizations.health.autoscaling_HorizontalPodAutoscaler: |
      hs = {}
      if obj.status ~= nil then
        if obj.status.conditions ~= nil then
          for i, condition in ipairs(obj.status.conditions) do
            if condition.type == "ScalingActive" and condition.reason == "FailedGetResourceMetric" then
                hs.status = "Progressing"
                hs.message = condition.message
                return hs
            end
            if condition.status == "True" then
                hs.status = "Healthy"
                hs.message = condition.message
                return hs
            end
          end
        end
        hs.status = "Healthy"
        return hs
      end
      hs.status = "Progressing"
      return hs
```

**Reply by prein** [reply]
@mubarak-j Thanks for sharing! I mentioned your solution in https://github.com/argoproj/argo-cd/issues/6287
